geode-protobuf/src/main/java/org/apache/geode/internal/protocol/protobuf/v1/authentication/AuthorizingCacheImpl.java

```diff
@@ -62,6 +70,7 @@ public AuthorizingCacheImpl(InternalCache cache, Authorizer authorizer) {
           authorizer.authorize(DATA, READ, regionName, key);
         }
         V value = (V) region.get(key);
+        value = (V) authorizer.postProcess(regionName, key, value);
```

Comment:
Should we be checking authorizer.needsPostProcessing() before calling this?

Reply:
Hmm, we could. I wasn't sure it would provide any performance benefit. The needsPostProcessing was mostly to avoid the complicated introspection and copying that's required for post processing oql query results.